### DISCUSSION TITLE
Implicit many-to-many custom table name with referentialIntegrity preview feature

### DISCUSSION THREAD

**Discussion by bs85** [prisma/prisma#12177] [Q&A]
I am working on a project running on PlanetScale where fkeys are not supported.

If I define an implicit many-to-many between say User and Role, it creates a _UserToRole pivot table automatically. We already have pivot tables (e.g. UserRoles), is there a way to keep the relationship implicit but point it to the right table?

**Comment by austincrim** [comment]
Hey @bs85 👋 ,

I believe if you follow all the [Prisma conventions](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations#conventions-for-relation-tables-in-implicit-m-n-relations) for the join table, it should work without an issue. I would go through the docs that I just linked and make sure your existing table follows what is outlined there. Then you should be able to run `prisma db pull` and have everything work as expected.

You can also edit the name of the underlying table using the `@relation` attribute.
```prisma
model Post {
  id         Int        @id @default(autoincrement())
  categories Category[] @relation("_MyRelationTable")
}

model Category {
  id    Int    @id @default(autoincrement())
  posts Post[] @relation("_MyRelationTable")
}
```

Let me know if you run into any trouble!

**Reply by janpio** [reply]
It might be that `referentialIntegrity` does not cover implicit m:n relations.

**Reply by vladi-stevanovic** [reply]
👋 Hello @bs85  I'm going through pending GitHub Discussions to check if you need any help. Is this issue still open?
Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! 🙌

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
